### ISSUE TITLE
HIFI Cryomag: Keithley2700 Temperature/Drift values

### ISSUE THREAD

**Issue by esouthren** [ISISComputingGroup/IBEX#2994]
As a user scientist, I would like to know the Temperature and Drift values, which are calculated from the returned Ohms resistance values. 

Temperature is calculated from a table based on the current Ohms value, and drift is then calculated based on a relation between temperature and time.

Acceptance criteria:

- Scanner display shows values of Temperature/Drift.

Note:

- PVs exists for CHNL:XXX:TEMP and :DRIFT but are unpopulated at the moment.

**Comment by esouthren** [comment]
This ticket calculates channel temperatures (CHNL:XXX:TEMP) by interpolating the CHNL:XXX:READ value from calibration files. 

The drift is then calculated via an aSub record which links to a C script (calculate_drift.c, in the support module). 


Pull requests - 

[Support module (aSub C script)](https://github.com/ISISComputingGroup/EPICS-KeithleyDVM/pull/3)
[OPI Changes](https://github.com/ISISComputingGroup/ibex_gui/pull/687)

In addition, pull from master in `/c/Instrument/Settings/config/common` to have the required`cryomagnet_scanner` folder, which contains 20 calibration files.

**Comment by esouthren** [comment]
Requires some unit tests for the new calculation functionality.

**Comment by LiamPanchaud** [comment]
PRs:

[OPI](https://github.com/ISISComputingGroup/ibex_gui/pull/687)

[IOC](https://github.com/ISISComputingGroup/EPICS-ioc/pull/270)

[Support](https://github.com/ISISComputingGroup/EPICS-KeithleyDVM/tree/Ticket2994_keithley_drift_value_c_script)

[Device Emulator](https://github.com/ISISComputingGroup/EPICS-DeviceEmulator/tree/Ticket2994_keithley_2700)

[IOC Tests](https://github.com/ISISComputingGroup/EPICS-IOC_Test_Framework/pull/142)

**Comment by LiamPanchaud** [comment]
Note that this is still in progress (on new branches), but most of the work is done, so the bulk is up for review

**Comment by aaron-long** [comment]
A tour de force in code. Nice work, please see the inline comments.

**Comment by aaron-long** [comment]
- Resolve merge conflict in Device Emulator -> command_builder.py
- Fix broken tests:

Devsim:
- test_GIVEN_empty_buffer_WHEN_reading_inserted_THEN_channel_PVs_get_correct_values
-  test_GIVEN_temperature_out_of_bounds_THEN_alarm_is_major

Devsim/Recsim:
- test_WHEN_buffer_control_set_THEN_buffer_control_matches_the_set_state (You renamed the control mode fields on the device)

I'm getting a persistent stream interface error in the devsim log:
````
2019-02-06 14:41:04,183 ERROR lewis.Keithley2700StreamInterface: An error occurred at request'TRAC:FREE?': AttributeError("'SimulatedKeithley2700' object has no attribute 'bytes_available'",)
2019-02-06 14:41:04,183 ERROR lewis.Keithley2700StreamInterface: Traceback (most recent call last):
  File "C:\Instrument\Apps\Python\lib\site-packages\lewis\adapters\stream.py", line 104, in found_terminator
    reply = cmd.process_request(request)
  File "C:\Instrument\Apps\Python\lib\site-packages\lewis\adapters\stream.py", line 366, in process_request
    return self.map_return_value(self.func(*args))
  File "C:\Instrument\Apps\EPICS\support\DeviceEmulator\master\lewis_emulators\keithley_2700\interfaces\stream_interface.py", line 219, in get_buffer_stats
    return "{}, {}".format(str(self._device.bytes_available), str(self._device.bytes_used))
AttributeError: 'SimulatedKeithley2700' object has no attribute 'bytes_available'
````

**Comment by LiamPanchaud** [comment]
I can't get those tests to fail, so either I fixed them a month ago and forgot, or it could be set-up dependent. Similarly, I don't see the `AttributeError` in the devsim lewis log anywhere. I do get a list out of bounds error, but that's expected given how the IOC works. 

Master has been merged into the branch of `ibex_gui`. 

Also note that 2 tests will fail if the reviewer has the most up-to-date version of `Settings\config\common\cryomagnet_scanner` because the calibration files have been flipped to work with ticket #3939 (the returned temperatures will be about 262K instead of the expected ~47K). The old calibration files are available somewhere else for a temporary replacement for testing.
